adala/runtimes/_openai.py

```diff
@@ -64,86 +54,53 @@ async def async_create_completion(
-            json={
-                "messages": messages,
-                "model": model,
-                "max_tokens": max_tokens,
-                "temperature": temperature,
-            },
-        ) as response:
-            response.raise_for_status()
-            response_json = await response.json()
-            completion_text = response_json["choices"][0]["message"]["content"]
-            return {
-                "text": completion_text,
-            }
-    except aiohttp.ClientResponseError as e:
-        # Handle HTTP errors
-        return {
-            "error": True,
-            "message": f"HTTP error: {e.status}",
-            "details": str(e),
-        }
-    except aiohttp.ClientError as e:
-        # Handle other aiohttp specific errors
-        return {
-            "error": True,
-            "message": "Client error",
-            "details": str(e),
-        }
-    except asyncio.TimeoutError as e:
-        # Handle timeout errors
+        completion = await client.chat.completions.create(
+            model=model,
+            messages=messages,
+            max_tokens=max_tokens,
+            temperature=temperature,
+        )
+        completion_text = completion.choices[0].message.content
         return {
-            "error": True,
-            "message": "Request timed out",
-            "details": str(e),
+            "text": completion_text,
+            "__adala_error__": False,
+            "__adala_message__": None,
+            "__adala_details__": None,
         }
     except Exception as e:
         # Handle other exceptions
         return {
-            "error": True,
-            "message": "Unknown error",
-            "details": str(e),
+            "text": None,
+            "__adala_error__": True,
+            "__adala_message__": "Unknown error",
+            "__adala_details__": str(e),
         }
 
 
 async def async_concurrent_create_completion(
     prompts,
-    max_concurrent_requests,
+    client,
     instruction_first,
     openai_model,
-    openai_api_key,
     max_tokens,
     temperature,
-    timeout=10,
 ):
-    semaphore = asyncio.Semaphore(max_concurrent_requests)
-
-    async with aiohttp.ClientSession(
-        timeout=aiohttp.ClientTimeout(total=timeout),
-    ) as session:
-        tasks = []
-        for prompt in prompts:
-            task = asyncio.ensure_future(
-                async_create_completion(
-                    user_prompt=prompt["user"],
-                    system_prompt=prompt["system"],
-                    semaphore=semaphore,
-                    session=session,
-                    model=openai_model,
-                    max_tokens=max_tokens,
-                    temperature=temperature,
-                    instruction_first=instruction_first,
-                    default_timeout=timeout,
-                    openai_api_key=openai_api_key,
-                )
+    tasks = []
+    for prompt in prompts:
```

Comment:
this loop seems simple enough that it may as well be a list comprehension. If you prefer the loop (maybe because of the large function signature), all good, but figured I'd suggest it.

Reply:
done, good call